Use sio2man/padman from ps2sdk

uyjulian opened the pull request:
## Pull Request checklist

Note: these are not necessarily requirements

- [x] I reformatted the code with clang-format
- [x] I checked to make sure my submission worked
- [x] I am the author of submission or have permission from the original author
- [ ] Requires update of the PS2SDK
- [ ] Requires update of the gsKit
- [ ] Others (please specify below)

## Pull Request description

Use sio2man/padman from ps2sdk

sp193 commented:
I attempted this before. If you do this, then you will break compatibility with homebrew that:
- Do not reboot the IOP and expect the necessary modules to be loaded.
- Are built to work with the default IOP modules. Particularly the original MCMAN, MCSERVE, SIO2MAN and PADMAN.

This is a problem for LaunchELF's current design, as the IOP modules used within the UI are also the ones that exist when a piece of software is loaded.

uyjulian commented:
I can probably add a IOP reboot with the old configuration before executing the ELF…

uyjulian commented:
Added compatible reset functionality.

AKuHAK commented:
Looks good to me. If I understand logic correctly, wLe itself will use modules from SDK. But when launching elf it will perform an additional reset which replicates the current wLe state.
@uyjulian 
Just to be sure - now ELF launch will take a longer time (as the reset function isn't that fast)? Did it break launch arguments when launching the new ELF?
Do you want to add something else in PR? It is already pretty old maybe you have new ideas?

AKuHAK reviewed (commented):
Looks good, however, it will be better to test the changes.

uyjulian commented:
I think I might attempt to move the module loading to before the screen is blanked, so the user knows that the IOP is reloading.

uyjulian commented:
I moved the reboot location to before the screen is cleared, and added an additional message specifying that the IOP is being rebooted.

However, I don't have hardware or the environment setup right now to test this functionality, so please check it.

AKuHAK commented:
@uyjulian console and pcsx2 just freeze after "Rebooting iop in compatibility mode."

uyjulian commented:
It looks like a ps2sdk update may have caused the black screen issue… I may bisect it later

uyjulian commented:
The issue has been found: https://github.com/ps2homebrew/wLaunchELF/pull/59

AKuHAK commented:
> The issue has been found: #59

However, this PR still produces a black screen.

uyjulian commented:
Since I added backwards compatible RPC to sio2man and padman to ps2sdk, I removed the reboot code.

AKuHAK commented:
dont know if we should try to fix ps2eth, or I just roll back github action to 1.3.0

uyjulian commented:
I already submitted a PR to fix ps2eth.
